### PR TITLE
Update maintenance window docs

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
Update maintenance window docs to include info about OS roll-outs.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: docs/usage/shoots.md [hunk 1]
```diff
@@ -40,13 +48,30 @@ spec:
...
+If you update the CloudProfile used in the Shoot and add `1.10.5` and `1.11.0` to the `.spec.<provider>.constraints.kubernetes.versions` list, the Shoot will be updated to `1.10.5` between 22:00-23:00 UTC. Your Shoot won't be updated to `1.11.0` even though its the highest Kubernetes in the CloudProfile, this is because that wouldn't be a patch release update but a minor release update, and potentially have breaking changes that could impact your deployed resources.
 
 In this example if the operator wants to update the Kubernetes version to `1.11.0`, he/she must update the Shoot's `.spec.kubernetes.version` to `1.11.0` manually.
 
+### Operating System Version
+
+If you update the `.spec.<provider>.constraints.machineImages` field in the CloudProfile used in the Shoot, then the maintenance controller will apply the new machine image to the Shoot spec (and will mark the Shoot to be reconciled) during the maintenance time window. During the reconciliation the corresponding `<Provider>MachineClass` resource in the Shoot namespace in the Seed will be updated and machine controller manager will take care the actual state to match the desired one.
```
Comment (plkokanov) [inline docs/usage/shoots.md]: Minor nit: "Seed will be updated and `the` machine controller manager will take care `of` the actual state to match the desired one.

File: docs/usage/shoots.md [hunk 2]
```diff
@@ -40,13 +48,30 @@ spec:
...
+
 # Configure a Shoot cluster alert receiver
+
 The receiver of the Shoot alerts can be configured by adding the annotation `garden.sapcloud.io/operatedBy` to the Shoot resource. The value of the annotation has to be a valid mail address.
 
 The alerting for the Shoot clusters is handled by the Prometheus Alertmanager. The Alertmanager will be deployed next to the control plane when the `Shoot` resource is annotated with the `garden.sapcloud.io/operatedBy` annotation and if a [SMTP secret](../deployment/configuration.md) exists.
 
-If the annotation gets removed then the Alertmanager will be also removed during the next reconcilation of the cluster. The same is valid in the opposite if the annotation is added to an existing cluster.
\ No newline at end of file
+If the annotation gets removed then the Alertmanager will be also removed during the next reconcilation of the cluster. The same is valid in the opposite if the annotation is added to an existing cluster.
```
Comment (plkokanov) [inline docs/usage/shoots.md]: Suggestion:
"The opposite is also valid if the annotation is added to an existing cluster"

File: pkg/apis/garden/validation/validation_test.go [hunk 3]
```diff
@@ -5595,7 +5595,7 @@ var _ = Describe("validation", func() {
 				},
 				Spec: garden.BackupInfrastructureSpec{
 					Seed:     "aws",
-					ShootUID: types.UID(utils.ComputeSHA1Hex([]byte(fmt.Sprintf(fmt.Sprintf("shoot-%s-%s", "garden", "backup-infrastructure"))))),
+					ShootUID: types.UID(utils.ComputeSHA1Hex([]byte(fmt.Sprintf("shoot-%s-%s", "garden", "backup-infrastructure")))),
```
Comment (plkokanov) [inline pkg/apis/garden/validation/validation_test.go]: Can't we just use:
```go
ShootUID: types.UID(utils.ComputeSHA1Hex([]byte("shoot-garden-backup-infrastructure))
```

**Review by plkokanov — CHANGES_REQUESTED** [review]

**Review by zanetworker — APPROVED** [review]
/lgtm

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**